examples/go/tagging/gotag-fn/handler.go

```diff
@@ -0,0 +1,177 @@
+package function
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"log"
+	"net/http"
+	"net/url"
+
+	handler "github.com/openfaas-incubator/go-function-sdk"
+	"github.com/pelletier/go-toml"
+	"github.com/vmware/govmomi"
+	"github.com/vmware/govmomi/vapi/rest"
+	"github.com/vmware/govmomi/vapi/tags"
+	"github.com/vmware/govmomi/vim25/types"
+)
+
+// tagInfo is part of the toml vcconfig file
+type tagInfo struct {
```

Comment:
could be simplified with

```go
type vcconfig struct {
	Tag struct {
		Action string
		URN    string
	}
	VCenter struct {
		Server   string
		Password string
		User     string
	}
}
```

Also see comments below on client handling - perhaps this struct could live in a separate client package and the logic be internalized

Reply:
Neat--I didn't know we could do that 😸